src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,320 @@
+
+    average = total_score / len(result.scores)
+    print("----------------------------")
+    print(f"Average Score:   {average:.2f}/5")
+    print(f"Total Score:     {total_score}/{len(result.scores)*5}\n")
+
+    print("Responses and scores written to:")
+    for file in results_files:
+        print(f"{file}")
+
+
+def create_excel_results_file(excel_file: str, result: EvaluationResult):
+    summary_df = pd.DataFrame()
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    summary_df["scores"] = scores
+
+    question_indices = [f"Q{i + 1}" for i in range(len(summary_df))]
+    question_indices.append("Average")
+    question_indices.append("Total Score")
+    question_indices.append("Median")
+
+    average = summary_df.mean(axis=0, numeric_only=True)
+    total_score = summary_df.sum(axis=0, numeric_only=True)
+    median = summary_df.median(axis=0, numeric_only=True)
+
+    summary_df.loc[len(summary_df)] = average
+    summary_df.loc[len(summary_df)] = total_score
+    summary_df.loc[len(summary_df)] = median
+
+    summary_df.insert(0, "question index", question_indices)
+
+    response_df = result.dataset.to_pandas()
+    response_df["scores"] = scores
+
+    with pd.ExcelWriter(excel_file) as writer:
+        response_df.to_excel(writer, sheet_name="dataset", index=False)
+        summary_df.to_excel(writer, sheet_name="Summary", index=False)
+    return summary_df
+
+
+def write_results(
+    result: EvaluationResult, file_formats: List[str], output_dir: str, model_name: str
+) -> List[str]:
+    validate_output_file_formats(file_formats)
+    response_df = result.dataset.to_pandas()
+    response_df["model_name"] = model_name
+
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    response_df["scores"] = scores
+
+    now = datetime.now()
+    timestamp = now.isoformat()
+    response_df["evaluation_run"] = f"run-{timestamp}"
+
+    results_files = []
+    for fmt in file_formats:
+        if IOFileType.JSONL.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.JSONL.value, output_dir, timestamp, model_name
+            )
+            response_df.to_json(f"{results_file}", orient="records", lines=True)
+            results_files.append(results_file)
+        elif IOFileType.CSV.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.CSV.value, output_dir, timestamp, model_name
+            )
+            response_df.to_csv(f"{results_file}", index=False)
+            results_files.append(results_file)
+        elif IOFileType.XLSX.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.XLSX.value, output_dir, timestamp, model_name
+            )
+            create_excel_results_file(results_file, result)
+            results_files.append(results_file)
+
+    return results_files
+
+
+def run_local_model_evaluation(
+    ctx,
```

Comment:
same as the other click comment, we should really avoid passing the `ctx` object or `click` specific things in general to avoid incurring tech debt for the click splitting.

eval has already been split, you can tell by the fact that there is both a `src/instructlab/model/evaluate.py` and a `src/instructlab/model/evaluate.py`, we cannot import the click library or use the ctx outside of the `cli` pkg

Reply:
Is the issue that `ctx` is being passed down or that `click` is being imported. I didn't realize that would be an issue since the `ctx` is being passed into `evaluate_data()` here: https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/evaluate.py#L33.

All of the uses of `ctx` are for the arguments passed into `launch_server` so a quick refactor should fix that.